### PR TITLE
Global config for bullet damage, speed multipliers

### PR DESCRIPTION
**Pull request by SecretAgent12**
https://trello.com/c/I1WCXFGX/453-global-config-for-bullet-damage-speed-multipliers

### CODE REVIEW

File: src/main/java/com/flansmod/common/FlansMod.java [hunk 1]
```diff
@@ -390,7 +392,6 @@ public void playerDrops(ItemTossEvent event) {
     @EventHandler
     public void registerCommand(FMLServerStartedEvent e) {
         CommandHandler handler = ((CommandHandler) FMLCommonHandler.instance().getSidedDelegate().getServer().getCommandManager());
-	handler.registerCommand(new CommandFlans());
```
Comment (GoldSloth) [inline src/main/java/com/flansmod/common/FlansMod.java:393]: Ehm.

File: src/main/java/com/flansmod/common/FlansMod.java [hunk 2]
```diff
@@ -668,7 +669,9 @@ public static void syncConfig() {
         TeamsManager.bulletSnapshotDivisor = configFile.getInt("BltSS_Divisor", "Teams/advanced settings", 50, 0, 1000, "Divisor");
 
         //Server/Gameplay Settings (Server-client synced)
-        enableKillMessages = configFile.getBoolean("enableKillMessages", "Gameplay Settings (synced)", enableKillMessages,"Enable killMessage display");
+        globalBulletSpeedMultiplier = configFile.getFloat("globalBulletSpeedMultiplier", "Gameplay Settings (synced)", 1.0F, 0F, 100F, "Set the global bullet speed multiplier");
+        globalWeaponDamageMultiplier = configFile.getFloat("globalWeaponDamageMultiplier", "Gameplay Settings (synced)", 1.0F, 0F, 100F, "Set the global weapon damage multiplier");
+        enableKillMessages = configFile.getBoolean("enableKillMessages", "Gameplay Settings (synced)", enableKillMessages,"Enable Enable killMessage display");
```
Comment (GoldSloth) [inline src/main/java/com/flansmod/common/FlansMod.java:674]: Could you rectify this pls?

File: src/main/java/com/flansmod/common/FlansMod.java [hunk 3]
```diff
@@ -668,7 +669,9 @@ public static void syncConfig() {
         TeamsManager.bulletSnapshotDivisor = configFile.getInt("BltSS_Divisor", "Teams/advanced settings", 50, 0, 1000, "Divisor");
 
         //Server/Gameplay Settings (Server-client synced)
-        enableKillMessages = configFile.getBoolean("enableKillMessages", "Gameplay Settings (synced)", enableKillMessages,"Enable killMessage display");
+        globalBulletSpeedMultiplier = configFile.getFloat("globalBulletSpeedMultiplier", "Gameplay Settings (synced)", 1.0F, 0F, 100F, "Set the global bullet speed multiplier");
```
Comment (GoldSloth) [inline src/main/java/com/flansmod/common/FlansMod.java:672]: All three of these also need to be added to PacketModConfig for syncing - including the kill message one

File: src/main/java/com/flansmod/common/guns/GunType.java [hunk 4]
```diff
@@ -500,7 +500,7 @@ protected void read(String[] split, TypeFile file) {
         super.read(split, file);
         try {
             if (split[0].equals("Damage"))
-                damage = Float.parseFloat(split[1]);
+                damage = Float.parseFloat(split[1])*FlansMod.globalWeaponDamageMultiplier;
```
Comment (GoldSloth) [inline src/main/java/com/flansmod/common/guns/GunType.java:503]: This cannot be done - on servers the multiplier is synced AFTER configs are read. IMO it needs to be applied when the damage variable is used (So in vehicle guns, ItemGun e.t.c.) And primary/secondary on vehicles that work with bullets directly, not guns.

File: src/main/java/com/flansmod/common/guns/GunType.java [hunk 5]
```diff
@@ -775,7 +775,7 @@ else if (split[0].equals("Ammo")) {
             } else if (split[0].equals("NumAmmoSlots") || split[0].equals("NumAmmoItemsInGun") || split[0].equals("LoadIntoGun"))
                 numPrimaryAmmoItems = Integer.parseInt(split[1]);
             else if (split[0].equals("BulletSpeed"))
-                bulletSpeed = Float.parseFloat(split[1]);
+                bulletSpeed = Float.parseFloat(split[1])*FlansMod.globalBulletSpeedMultiplier;
```
Comment (GoldSloth) [inline src/main/java/com/flansmod/common/guns/GunType.java:778]: Same here, this isn't going to work very well I'm afraid :(

**Review by GoldSloth — CHANGES_REQUESTED** [review]

File: src/main/java/com/flansmod/common/FlansMod.java (hunk 2 above)
Comment (SecretAgent12) [reply src/main/java/com/flansmod/common/FlansMod.java:674]: yes

**Pull request closed without merging**